### PR TITLE
[Accelerator 4] Adding Consent Management Implementation

### PR DESCRIPTION
**Pull request by Ashi1993**
## Adding Consent Management Implementation

> This PR adds the Consent Management Implementation.

**Issue link:** *https://github.com/wso2-enterprise/solutions-bfsi/issues/20*

**Doc Issue:** *Optional, link issue from [documentation repository](https://github.com/wso2/docs-ob/issues)*

**Applicable Labels:** *Spec, product, version, type (specify requested labels)*

------

### Development Checklist

1. [x] Build complete solution with pull request in place.
2. [x] Ran checkstyle plugin with pull request in place.
3. [x] Ran Findbugs plugin with pull request in place.
4. [x] Ran FindSecurityBugs plugin and verified report.
5. [x] Formatted code according to WSO2 code style.
6. [x] Have you verified the PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?
7. [ ] Migration scripts written (if applicable).
8. [x] Have you followed secure coding standards in [WSO2 Secure Engineering Guidelines](http://wso2.com/technical-reports/wso2-secure-engineering-guidelines)?

### Testing Checklist

1. [x] Written unit tests.
2. [ ] Verified tests in multiple database environments (if applicable).
3. [ ] Tested with BI enabled  (if applicable).

### CODE REVIEW

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/constant/FinancialServicesConstants.java [hunk 1]
```diff
@@ -269,5 +266,24 @@ public class FinancialServicesConstants {
     public static final String DOT_SEPARATOR = ".";
     public static final String PRODUCTION = "PRODUCTION";
     public static final String SANDBOX = "SANDBOX";
+
+// NEw
```
Comment (RivinduM) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/constant/FinancialServicesConstants.java:270]: typo

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.dao/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/dao/constants/ConsentMgtDAOConstants.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    public static final String WHERE = "where";
+    public static final String PLACEHOLDER = "placeholder";
+    public static final String PLAIN_PLACEHOLDER = "plainPlaceholder";
+    public static final String EQUALS = "equals";
+    // Consent Database Table Identifiers
+    public static final String TABLE_OB_CONSENT = "OB_CONSENT";
+    public static final String TABLE_OB_CONSENT_AUTH_RESOURCE = "OB_CONSENT_AUTH_RESOURCE";
+    public static final String TABLE_OB_CONSENT_MAPPING = "OB_CONSENT_MAPPING";
+    public static final String TABLE_OB_CONSENT_ATTRIBUTE = "OB_CONSENT_ATTRIBUTE";
+    public static final String TABLE_OB_CONSENT_FILE = "OB_CONSENT_FILE";
```
Comment (RivinduM) [inline financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.dao/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/dao/constants/ConsentMgtDAOConstants.java:87]: Are we still using the OB prefixes?

Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.consent.mgt.dao/src/main/java/org/wso2/financial/services/accelerator/consent/mgt/dao/constants/ConsentMgtDAOConstants.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/108/commits/74e401a23f66920af64aee5717dd4ccc8a075948

File: financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/constant/FinancialServicesConstants.java (hunk 1 above)
Comment (Ashi1993) [reply financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/constant/FinancialServicesConstants.java]: FIxed in https://github.com/wso2/financial-services-accelerator/pull/108/commits/74e401a23f66920af64aee5717dd4ccc8a075948

**Review by RivinduM — APPROVED** [review]

**Review by VimukthiRajapaksha — APPROVED** [review]

**Pull request merged**